cloudify_agent/worker.py

```diff
@@ -160,9 +163,16 @@ def cluster_update_task(self, nodes):
         daemon.cluster = nodes
         factory.save(daemon)
 
-    def cancel_workflow_task(self, execution_id):
+    def cancel_workflow_task(self, execution_id, rest_token, tenant):
         logger.info('Cancelling execution {0}'.format(execution_id))
+
+        class CancelCloudifyContext(object):
+            def __init__(self):
+                self.tenant = tenant['name']
+                self.rest_token = rest_token
         self._workflow_registry.cancel(execution_id)
```

Comment:
Can `cancel` fail? If it does, don't we still want to update the execution status?

Reply:
I think it has no real failure modes, no. So if it does fail and the programmer says it shouldn't, it's not safe to continue anyway ;)